source/plot/Axis.ooc

```diff
@@ -42,56 +42,63 @@ Axis: class {
+				numberOffset := FloatPoint2D new(0.0f, 3.0f + (this fontSize - 4))
+				tickMarkerEndOffset := FloatPoint2D new(0.0f, -5.0f)
+				topTickMarkerStartOffset := FloatPoint2D new(0.0f, - plotAreaSize y)
+				result = result & Shapes text(position + labelOffset, this label, this fontSize, "middle")
+				if (radix >= pow(10, this precision - 1) || radix <= pow(10, - this precision)) {
+					radixOffset := FloatPoint2D new(plotAreaSize x + margin x / 2.0f, numberOffset y)
+					scientificPower := Float getScientificPowerString(radix)
+					result = result & Shapes text(position + radixOffset, scientificPower, this fontSize, "middle")
+					scientificPower free()
+				}
+
+				tickValue := this getFirstTickValue()
+				position x += translationToOrigo x + scaling x * tickValue
+				while (tickValue <= this max) {
+					result = result & this getTickSVG(tickValue, radix, position, numberOffset, topTickMarkerStartOffset, tickMarkerEndOffset, "middle")
+					tickValue += this tick
+					position x += scaling x * tick
+				}
+			} else {
+				result = result >> "<g desc='Y-axis data'>\n"
+				labelOffset := FloatPoint2D new(- (log10(Float maximum(this max, Float absolute(this min)) / radix) + 3.0f) * 0.6f * (this fontSize - 4.0f), - plotAreaSize y / 2.0f)
+				numberOffset := FloatPoint2D new(- 0.6f * (this fontSize - 4.0f), (this fontSize - 4.0f) / 3.0f)
+				tickMarkerEndOffset := FloatPoint2D new(5.0f, 0.0f)
+				rightTickMarkerStartOffset := FloatPoint2D new(plotAreaSize x, 0.0f)
+				result = result >> "<g desc='rotated Y-axis' transform='rotate(-90," & (position x + labelOffset x) toString() >> "," & (position y + labelOffset y) toString() >> ")'>\n"
+				result = result & Shapes text(position + labelOffset, this label, this fontSize, "middle")
+				result = result >> "</g>\n"
+				if (radix >= pow(10, this precision - 1) || radix <= pow(10, - this precision)) {
+					radixOffset := FloatPoint2D new(numberOffset x, - plotAreaSize y - margin y / 2 + this fontSize / 3.0f)
+					scientificPower := Float getScientificPowerString(radix)
+					result = result & Shapes text(position + radixOffset, scientificPower, this fontSize, "end")
+					scientificPower free()
+				}
 
-			tickValue := this getFirstTickValue()
-			position y += translationToOrigo y - plotAreaSize y - scaling y * tickValue
-			while (tickValue <= this max) {
-				result = result & this getTickSVG(tickValue, radix, position, numberOffset, rightTickMarkerStartOffset, tickMarkerEndOffset, "end")
-				tickValue += this tick
-				position y += - scaling y * tick
+				tickValue := this getFirstTickValue()
+				position y += translationToOrigo y - plotAreaSize y - scaling y * tickValue
+				while (tickValue <= this max) {
+					result = result & this getTickSVG(tickValue, radix, position, numberOffset, rightTickMarkerStartOffset, tickMarkerEndOffset, "end")
+					tickValue += this tick
+					position y -= scaling y * tick
+				}
 			}
+			result = result >> "</g>\n"
 		}
```

Comment:
This function is too long and contains a lot of duplicate code. Instead of a huge if/else block, consider adding the functions `getHorizontalSvg` and `getVerticalSvg` or something and try to move any duplicate code into their own functions.